test/torchaudio_unittest/rnnt/autograd_impl.py

```diff
@@ -0,0 +1,53 @@
+from typing import Callable, Tuple
+import torch
+from torch import Tensor
+from torch.autograd import gradcheck
+from torchaudio_unittest.common_utils import (
+    TestBaseMixin,
+)
+from .utils import get_numpy_random_data, numpy_to_torch, get_data_basic
+from torchaudio.prototype.rnnt_loss import RNNTLoss, rnnt_loss
+
+class Autograd(TestBaseMixin):
+    def assert_grad(
+            self,
+            transform: Callable[..., Tensor],
+            inputs: Tuple[torch.Tensor],
+            *,
+            enable_all_grad: bool = True,
+    ):
+        inputs_ = []
+        for i in inputs:
+            if torch.is_tensor(i):
+                i = i.to(dtype=self.dtype, device=self.device)
+                if enable_all_grad:
+                    i.requires_grad = True
+            inputs_.append(i)
+        assert gradcheck(transform, inputs_, eps=1e-06, atol=1e-05)
+
+    def test_rnnt_loss_small(self):
+        inputs = get_data_basic(self.device)
```

Comment:
@vincentqb 
using an example from warp-transducer instead, I get the following:
**torchaudio rnnt**
`numerical:tensor([[-3.4612e-304],
[  0.0000e+00],
[  0.0000e+00],
[-1.1240e-312],
[-2.3224e-304],
[ 2.3224e-304],
[-3.4612e-304],
[  0.0000e+00],
[-2.3224e-304],
[  0.0000e+00],
[  0.0000e+00],
[  0.0000e+00],
[  0.0000e+00],
[ 2.3224e-304],
[-2.3224e-304],
[  0.0000e+00],
[-3.3710e+205],
[-3.4612e-304],
[-2.3224e-304],
[-2.3224e-304],
[ 2.3224e-304],
[  0.0000e+00],
[ 2.3224e-304],
[  0.0000e+00],
[-2.3224e-304],
[ 2.3224e-304],
[  0.0000e+00],
[-7.9051e-317],
[-1.1388e-304],
[  0.0000e+00]], dtype=torch.float64)`
`analytical:tensor([[0.1000],
[0.6000],
[0.1000],
[0.1000],
[0.1000],
[0.1000],
[0.1000],
[0.6000],
[0.1000],
[0.1000],
[0.1000],
[0.1000],
[0.2000],
[0.8000],
[0.1000],
[0.1000],
[0.6000],
[0.1000],
[0.1000],
[0.1000],
[0.1000],
[0.1000],
[0.2000],
[0.1000],
[0.1000],
[0.7000],
[0.1000],
[0.2000],
[0.1000],
[0.1000]], dtype=torch.float64, grad_fn=<CopySlices>)`

**warp-transducer**
`numerical:tensor([[ 0.0000],
[-0.4768],
[ 0.2384],
[ 0.2384],
[ 0.2384],
[ 0.0000],
[ 0.2384],
[-0.2384],
[ 0.2384],
[ 0.2384],
[-0.2384],
[ 0.2384],
[ 0.2384],
[ 0.2384],
[ 0.2384],
[ 0.0000],
[ 0.0000],
[ 0.0000],
[ 0.0000],
[ 0.0000],
[ 0.0000],
[ 0.0000],
[-0.4768],
[ 0.0000],
[ 0.0000],
[-0.7153],
[ 0.2384],
[ 0.2384],
[ 0.2384],
[ 0.2384]])`
`analytical:tensor([[-0.1312],
[-0.3999],
[ 0.1770],
[ 0.1770],
[ 0.1770],
[-0.1857],
[ 0.1225],
[-0.1817],
[ 0.1225],
[ 0.1225],
[-0.3209],
[ 0.0627],
[ 0.0693],
[ 0.1262],
[ 0.0627],
[ 0.0546],
[-0.2182],
[ 0.0546],
[ 0.0546],
[ 0.0546],
[ 0.1207],
[ 0.1207],
[-0.4830],
[ 0.1207],
[ 0.1207],
[-0.6926],
[ 0.1687],
[ 0.1865],
[ 0.1687],
[ 0.1687]])`

Reply:
I think the reason for analytical == logits for torchaudio might be the use of float64, since only float16 and float32 should be supported. Using float 32 results in the following instead
`numerical:tensor([[ 1.5268e+05],
[ 1.3338e+04],
[ 4.2081e+02],
[ 8.6069e+01],
[-6.2895e+02],
[ 5.2094e+02],
[ 1.8239e+02],
[ 5.1594e+02],
[-7.3195e+01],
[-1.2159e+01],
[ 4.6015e+01],
[ 3.7432e+01],
[ 3.1948e+01],
[ 2.7657e+01],
[ 1.4544e+02],
[-3.4332e+01],
[-6.1989e+01],
[ 2.6464e+01],
[ 3.8147e+00],
[ 2.6226e+00],
[ 4.7684e-01],
[ 7.1526e-01],
[-3.8624e+01],
[ 8.5831e+00],
[ 4.7684e-01],
[ 9.5367e-01],
[ 9.5367e-01],
[ 9.5367e-01],
[ 9.5367e-01],
[-2.3842e-01]])`
`analytical:tensor([[ 0.2438],
[-0.5317],
[ 0.2438],
[ 0.2438],
[-0.1996],
[ 0.1468],
[ 0.1468],
[-0.2588],
[ 0.1468],
[-0.1816],
[ 0.0762],
[ 0.0762],
[ 0.0762],
[ 0.0762],
[-0.3046],
[ 0.0760],
[-0.3041],
[ 0.0760],
[ 0.0760],
[ 0.0760],
[ 0.1433],
[ 0.1433],
[-0.5734],
[ 0.1433],
[ 0.1433],
[ 0.2318],
[ 0.2318],
[ 0.2318],
[ 0.2318],
[-0.9273]], grad_fn=<CopySlices>)`
Additionally, these numbers match with running gradcheck on the internal version as well